### PR TITLE
feat(list): add OrderedDisplayList component

### PR DESCRIPTION
**Pull request by TheSisb**
Implements new exports to the list package: `OrderedDisplayList`, `OrderedDisplayListItem`, and `OrderedDisplayListHeading`.

[Figma](https://www.figma.com/file/l2qtpQVMmXzQPVB3rEAA8i/Paste-Components-%26-Themes?node-id=6560%3A54907&mode=dev)

### CODE REVIEW

File: packages/paste-core/components/inline-code/src/InlineCode.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ export interface InlineCodeProps extends Partial<Omit<HTMLPasteProps<"div">, "ch
    */
   element?: BoxProps["element"];
   /**
-   * Only the "default" variant should be used as the "minimal" variant is deprecated.
+   * Different inline code block styles. Defaults to "default".
```
Comment (nkrantz) [inline packages/paste-core/components/inline-code/src/InlineCode.tsx:23]: question: why did you remove the deprecation note

File: packages/paste-core/components/list/__tests__/index.spec.tsx [hunk 2]
```diff
@@ -267,3 +238,143 @@ describe("ListItem", () => {
...
+            Text
+          </OrderedDisplayListItem>
+        </OrderedDisplayList>,
+      );
+      expect(screen.getByRole("list").getAttribute("data-paste-element")).toEqual("foo");
+      expect(screen.getByRole("listitem").getAttribute("data-paste-element")).toEqual("fooitem");
+      expect(screen.getByRole("heading").getAttribute("data-paste-element")).toEqual("foohead");
+    });
+
+    it("should add custom styles to OrderedDisplayList", (): void => {
```
Comment (nkrantz) [inline packages/paste-core/components/list/__tests__/index.spec.tsx:311]: didn't you prove that we technically don't need these tests with customization VRT?

File: packages/paste-core/components/list/src/types.ts [hunk 3]
```diff
@@ -45,5 +46,15 @@ export type ListItemProps = HTMLPasteProps<"li"> & {
 };
 
 export type OrderedListProps = Omit<BaseListProps, "as" | "listStyleType">;
-
 export type UnorderedListProps = Omit<BaseListProps, "as" | "listStyleType">;
+
+// Ordered Display List
+export type OrderedDisplayListVariants = "heading20" | "heading30" | "heading40" | "heading50" | "heading60";
```
Comment (nkrantz) [inline packages/paste-core/components/list/src/types.ts:52]: very non-blocking
```suggestion
export type OrderedDisplayListVariants = Omit<HeadingVariants, "heading10">;
```

File: packages/paste-core/components/list/src/display/OrderedDisplayListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  ({ children, element = "ORDERED_DISPLAY_LIST_ITEM", ...props }, ref) => {
+    const { variant } = React.useContext(OrderedDisplayListContext);
+
+    /*
+     * We need to wrap children in a div in order to apply the grid styles correctly when there are n children.
+     * The div keeps children's count to 1.
+     */
+    return (
+      <StyledOrderedListItem {...safelySpreadBoxProps(props)} as="li" element={element} variant={variant} ref={ref}>
+        <div>{children}</div>
```
Comment (nkrantz) [inline packages/paste-core/components/list/src/display/OrderedDisplayListItem.tsx:99]: suggestion: use a Box?

File: packages/paste-core/components/list/src/display/OrderedDisplayListContext.tsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import * as React from "react";
+
+import type { OrderedDisplayListVariants } from "../types";
+
+interface OrderedDisplayListState {
+  variant: OrderedDisplayListVariants;
+}
+
+export const OrderedDisplayListContext = React.createContext<OrderedDisplayListState>({ variant: "heading20" });
```
Comment (nkrantz) [inline packages/paste-core/components/list/src/display/OrderedDisplayListContext.tsx:9]: should this be updated to `heading30` default?

File: packages/paste-core/components/list/__tests__/index.spec.tsx (hunk 2 above)
Comment (TheSisb) [reply packages/paste-core/components/list/__tests__/index.spec.tsx:311]: I forgot I did that, been a while. Removing lol

File: packages/paste-core/components/inline-code/src/InlineCode.tsx (hunk 1 above)
Comment (TheSisb) [reply packages/paste-core/components/inline-code/src/InlineCode.tsx:23]: didnt mean to check that in :X

File: packages/paste-core/components/list/src/display/OrderedDisplayListItem.tsx (hunk 4 above)
Comment (TheSisb) [reply packages/paste-core/components/list/src/display/OrderedDisplayListItem.tsx:99]: Box isn't needed, adds unnecessary react overhead here

File: packages/paste-core/components/list/src/display/OrderedDisplayListContext.tsx (hunk 5 above)
Comment (TheSisb) [reply packages/paste-core/components/list/src/display/OrderedDisplayListContext.tsx:9]: Good catch

File: packages/paste-core/components/list/src/types.ts (hunk 3 above)
Comment (TheSisb) [reply packages/paste-core/components/list/src/types.ts:52]: Good suggestion!

**Review by roxanin — APPROVED** [review]
